### DISCUSSION TITLE
JMeter remote testing support

### DISCUSSION THREAD

**Discussion by mricciuti** [qualersoft/jmeter-gradle-plugin#49] [Ideas]
Hi

Does this plugin support JMeter remote test feature ? (https://jmeter.apache.org/usermanual/remote-test.html)
I can see that JMeter extension allows us to set global properties ( -G flag), which are supposed to be sent to each JMeter remote agent, but how can we configure the list of remote servers?  I don't see anything in the task or extension code that would set the `-R` jmeter command line argument.

This feature is quite useful, and is supported by other jmeter plugins like this Maven plugin: https://github.com/jmeter-maven-plugin/jmeter-maven-plugin/wiki/Remote-Server-Configuration

I think this could be done by adding some dedicated properties in the JMeter extension to define the servers list plus some additional options, and configure the `-R` flag accordingly.

Thanks by advance, and keep the good work ;)

**Comment by mricciuti** [comment]
Hi,

I had a deeper look, and it seems quite easy to make this plugin support the remote testing mode with very little changes:
- the target servers list can already be set using the jmeter property `remote_hosts` that we can configure on  `jmeter.jmeterProperties` map,  => OK
- global properties (sent to each target node) is already available (=> -G argument) => OK
- ssl configuration: I made some tests with rmi ssl disabled, which can be configured with jmeter property `server.rmi.ssl.disable` => OK

The only thing missing in the plugin itself is the possibility to set the jmeter `-r` flag to trigger remote execution:  this flag could be handled in the `JMeterRunTask.createRunArguments()` method, based a new property (e.g. "remoteExecution: true/false') in the jmeter extension.

What do you think ?

**Comment by mathze** [comment]
Hi @mricciuti,

first of all sorry for late reply.
That said, originally it was by design to not provide any functionality related to remoting. I was - and still am :) - afraid of creating false expectations for the plugin.
Never the less, just routing an additional argument through the plugin is not much effort but I can see the big value this will bring to users.
Thank you very much for the feedback and your great analysis. 👍
